modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/application/MuleApplicationPolicyProvider.java

```diff
@@ -108,15 +111,15 @@ public synchronized boolean removePolicy(String parametrizedPolicyId) {
   }
 
   @Override
-  public List<org.mule.runtime.core.policy.Policy> findSourceParameterizedPolicies(
-                                                                                   PolicyPointcutParameters policyPointcutParameters) {
-    List<org.mule.runtime.core.policy.Policy> policies = new ArrayList<>();
+  public List<Policy> findSourceParameterizedPolicies(PolicyPointcutParameters policyPointcutParameters) {
+    List<Policy> policies = new ArrayList<>();
 
     if (!registeredPolicyInstanceProviders.isEmpty()) {
       for (RegisteredPolicyInstanceProvider registeredPolicyInstanceProvider : registeredPolicyInstanceProviders) {
-        if (registeredPolicyInstanceProvider.policyInstanceProvider.getPointcut().matches(policyPointcutParameters)) {
-          policies.addAll(registeredPolicyInstanceProvider.policyInstanceProvider
-              .findSourceParameterizedPolicies(policyPointcutParameters));
+        if (registeredPolicyInstanceProvider.applicationPolicyInstance.getPointcut().matches(policyPointcutParameters)) {
+          if (registeredPolicyInstanceProvider.applicationPolicyInstance.getSourcePolicy().isPresent()) {
```

Comment:
the instance should be exposed through a getter instead of a field directly. This is hard to mock this way